pkg/capr/common.go

```diff
@@ -674,3 +674,54 @@ func PreBootstrap(mgmtCluster *v3.Cluster) bool {
 
 	return !v3.ClusterConditionPreBootstrapped.IsTrue(mgmtCluster)
 }
+
+// FormatWindowsEnvVar accepts a corev1.EnvVar and returns a string to be used in either
+// a Powershell script or the Rancher planner, indicated by the isPlanVariable parameter.
+// This function automatically configures the '$env:' prefix for a given environment variable,
+// automatically prefixes boolean values with '$', and surrounds string variables with double quotes as
+// needed. If the provided variable name incorrectly uses either '$env:' or '$' for the given isPlanVariable
+// value, it will be removed.
+func FormatWindowsEnvVar(envVar corev1.EnvVar, isPlanVariable bool) string {
```

Comment:
IMO we should try to refactor this to use the `rkev1.EnvVar` if we can since that is what is on the control plane, not sure if its doable but that way you won't have to rewrap them everywhere.

Reply:
I had debated between the two, but since `WindowsInstallScript` already uses `corev1` env vars I went with that type. Right now we are only wrapping a `rkev1.EnvVar` into a `corev1.EnVar` in the `generateInstallInstruction` function when passing the agent env vars into the plan. Everywhere else we are either already working with a `corev1.EnvVar` or creating one based off of simple strings. The difference isn't a big deal to me, if we think it will be more useful to move to `rkev1.EnvVar` I can make that change

Reply:
Eh just merge it, looking at the code it's probably tech debt so no point in blocking your PR to fix it. I'll probably try to fix it at some point in the future.